### DISCUSSION TITLE
How to change the name of the 404 page in Nuxt3

### DISCUSSION THREAD

**Discussion by Red-Asuka** [nuxt/nuxt#26394] [Questions]
In Nuxt2, we can customize the file name of the 404 page through the following configuration, but I did not find a similar configuration in Nuxt3. How can I achieve similar functionality?

```ts
  generate: {
    fallback: 'custom-404.html',
  },
```

**Comment by Red-Asuka** [comment]
I found the relevant code in the source code, but I couldn't find a way to modify it. I tried to add it to `nitro:config` hook in `config.prerender?.routes`, but this resulted in an error during prerendering. And I have no way to delete the pre-rendering of the default `/404.html`.

<img width="971" alt="image" src="https://github.com/nuxt/nuxt/assets/38158783/fc3aa539-fc00-46a6-a087-d4c713d1acc9">

```ts
'nitro:config': async (config) => {
  // https://github.com/nuxt/nuxt/issues/13949#issuecomment-1806609372
  if (!process.argv?.includes('generate'))
    return

  config.prerender?.routes?.push('/custom-404.html')
},
```

<img width="644" alt="image" src="https://github.com/nuxt/nuxt/assets/38158783/14f0997a-3e6e-4645-a32d-e3a7406c0613">

**Reply by xjccc** [reply]
nuxt.config.ts
like this? It will add a routes, but not change the name.
```
  hooks: {
    'prerender:routes'({ routes }) {
      routes.add('/custom-404.html')
    }
  }
```

**Reply by Red-Asuka** [reply]
This will display the same prerendering error.

**Reply by xjccc** [reply]
Yes. i'm not found how to change this.😂
